tests/test_authentication.py

```diff
@@ -57,9 +65,20 @@ def test_get_initial_tokens_with_credentials_from_env_variables(base_url, creden
     unstub()
 
 
+def test_get_initial_tokens_with_incomplete_credentials_from_env_variables(base_url, credentials, monkeypatch):
+    """
+    Tests configuration error is reported if IQM_AUTH_SERVER is set, but no credentials provided
+    """
+    monkeypatch.setenv('IQM_AUTH_SERVER', credentials['auth_server_url'])
+    with pytest.raises(ClientConfigurationError) as e:
+        IQMClient(base_url)
+    assert str(e.value) == 'Auth server URL is set but no username or password'
+    unstub()
+
+
 def test_add_authorization_header_when_credentials_are_provided(base_url, credentials):
     """
-    Tests that requests are sent with Authorization header when credentials are provided
+    Tests that `get_run` requests are sent with Authorization header when credentials are provided
```

Comment:
```suggestion
    Tests that ``get_run`` requests are sent with Authorization header when credentials are provided
```

I can't remember the exact edge cases, but we use double backticks in Markdown... I think it's for better docs rendering (when inline code is placed inside a docstring or something)